source/localizable/tutorial/autocomplete-component.md

```diff
@@ -1,100 +1,173 @@
-property in our component. Clicking the list item will fire the `choose`
-action with the `city` property of the item as a parameter, which will
-then populate the `input` field with the name of that `city`.
+```tests/integration/components/list-filter-test.js
+import Ember from 'ember';
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import wait from 'ember-test-helpers/wait';
 
-Here is what the component's JavaScript looks like:
+moduleForComponent('list-filter', 'Integration | Component | filter listing', {
+  integration: true
+});
 
-```app/components/filter-listing.js
-export default Ember.Component.extend({
-  filter: null,
+const ITEMS = [{city: 'San Francisco'}, {city: 'Portland'}, {city: 'Seattle'}];
+const FILTERED_ITEMS = [{city: 'San Francisco'}];
 
-  actions: {
-    choose(city) {
-      this.set('filter', city);
+test('should initially load all listings', function (assert) {
+  // we want our actions to return promises, since they are potentially fetching data asynchronously
+  this.on('filterByCity', (val) => {
+    if (val === '') {
+      return Ember.RSVP.resolve(ITEMS);
+    } else {
+      return Ember.RSVP.resolve(FILTERED_ITEMS);
     }
-  }
+  });
+  
+  // with an integration test, you can set up and use your component in the same way your application 
+  // will use it.
+  this.render(hbs`
+    {{#list-filter filter=(action 'filterByCity') as |rentals|}}
+      <ul>
+      {{#each rentals as |item|}}
+        <li class="city">
+          {{item.city}}
+        </li>
+      {{/each}}
+      </ul>
+    {{/list-filter}}
+  `);
+
+  // the wait function will return a promise that will wait for all promises 
+  // and xhr requests to resolve before running the contents of the then block.
+  return wait().then(() => {
+    assert.equal(this.$('.city').length, 3);
+    assert.equal(this.$('.city').first().text().trim(), 'San Francisco');
+  });
 });
 ```
+For our second test we'll check that typing text in the filter will actually appropriately call the filter action and update the listings shown.
+
+We force the action by generating a `keyUp` event on our input field, and then assert that only one item is rendered.
+
+```tests/integration/components/list-filter-test.js
+test('should update with matching listings', function (assert) {
+  this.on('filterByCity', (val) => {
+    if (val === '') {
+      return Ember.RSVP.resolve(ITEMS);
+    } else {
+      return Ember.RSVP.resolve(FILTERED_ITEMS);
+    }
+  });
 
-There's a property for each of the `filter` and `filteredList`, and
-the `choose` action as described above.
+  this.render(hbs`
+    {{#list-filter filter=(action 'filterByCity') as |rentals|}}
+      <ul>
+      {{#each rentals as |item|}}
+        <li class="city">
+          {{item.city}}
+        </li>
+      {{/each}}
+      </ul>
+    {{/list-filter}}
+  `);
+
+  this.$('.list-filter input').val('San').keyup();
```

Comment:
Maybe add a comment above the keyup?  Took me a bit looking at this test to find the difference ;)